src/store/ChartState.js

```diff
@@ -75,101 +76,70 @@ class ChartState {
         scrollToEpoch,
         scrollToEpochOffset = 0,
         settings,
+        shouldFetchTradingTimes = true,
         showLastDigitStats = false,
         startEpoch,
         symbol,
         zoom,
-        shouldFetchTradingTimes = true,
     }) {
         this.chartId = id;
-        this.settings = settings;
-        this.isConnectionOpened = isConnectionOpened;
         this.chartStatusListener = chartStatusListener;
+        this.isAnimationEnabled = isAnimationEnabled;
+        this.isConnectionOpened = isConnectionOpened;
         this.isStaticChart = isStaticChart;
+        this.scrollToEpochOffset = scrollToEpochOffset;
+        this.settings = settings;
         this.shouldFetchTradingTimes = shouldFetchTradingTimes;
+        this.showLastDigitStats = showLastDigitStats;
 
-        if (this.symbol !== symbol) {
-            this.symbol = symbol;
+        this.startEpoch = startEpoch;
+        this.endEpoch = endEpoch;
+        this.margin = margin;
 
+        // IMPORTANT! Set chartType first then set chart granularity
+        this.setChartType(chartType);
```

Comment:
@kellybinary we can move these to function into the switch's body but without break.

Reply:
@morteza-binary sure 👍